File: root/programs/C#/Frameworks/Infrastructure/Business/AsyncProcessingService/AsyncProcessingServiceParameterValue.cs
//**********************************************************************************
//* Copyright (C) 2007,2016 Hitachi Solutions,Ltd.
//**********************************************************************************

#region Apache License
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License. 
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
#endregion

//**********************************************************************************
//* クラス名            :AsyncProcessingServiceParameterValue.cs
//* クラス名クラス名     :
//*
//* 作成者              :Supragyan
//* クラス日本語名       :
//* 更新履歴
//*  Date:        Author:        Comments:
//*  ----------  ----------------  -------------------------------------------------
//*  11/28/2014   Supragyan      Paramter Value class for Asynchronous Processing Service
//*  04/15/2015   Sandeep        Changed datatype of ProgressRate to decimal.
//**********************************************************************************
// System
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

//業務フレームワーク
using Touryo.Infrastructure.Business.Common;
using Touryo.Infrastructure.Business.Util;
using System.Reflection;

namespace AsyncProcessingService
{
    /// <summary>
    /// Paramter Value class for Asynchronous Processing Service
    /// </summary>
    public class AsyncProcessingServiceParameterValue : MyParameterValue
    {
        /// <summary>汎用エリア</summary>
        public object Obj;

        /// <summary>TaskId</summary>
        public int TaskId;

        /// <summary>UserId</summary>
        public string UserId;

        /// <summary>ProcessName</summary>
        public string ProcessName;

        /// <summary>Data</summary>
        public string Data;

        /// <summary>RegistrationDateTime</summary>
        public DateTime RegistrationDateTime;

        /// <summary>ExecutionStartDateTime</summary>
        public DateTime ExecutionStartDateTime;

        /// <summary>NumberOfRetries</summary>
        public int NumberOfRetries;

        /// <summary>ProgressRate</summary>
        public decimal ProgressRate;

        /// <summary>Status</summary>
        public int StatusId;

        /// <summary>CompletionDateTime</summary>
        public DateTime CompletionDateTime;

        /// <summary>CommandId</summary>
        public int CommandId;

        /// <summary>ReservedArea</summary>
        public string ReservedArea;

        /// <summary>ExceptionInfo</summary>
        public string ExceptionInfo;

        #region コンストラクタ

        /// <summary>コンストラクタ</summary>
        public AsyncProcessingServiceParameterValue(string screenId, string controlId, string methodName, string actionType, MyUserInfo user)
            : base(screenId, controlId, methodName, actionType, user)
        {
            // Baseのコンストラクタに引数を渡すために必要。
        }

        #endregion

        #region AsyncStatus

        /// <summary>
        /// AsyncStatus Enum for storing all status
        /// </summary>
        public enum AsyncStatus
        {
            /// <summary>Register</summary>
            [StringValue("Register")]
            Register = 1,

            /// <summary>Processing</summary>
            [StringValue("Processing")]
            Processing,

            /// <summary>End</summary>
            [StringValue("End")]
            End,

            /// <summary>AbnormalEnd</summary>
            [StringValue("AbnormalEnd")]
            AbnormalEnd,

            /// <summary>Abort</summary>
            [StringValue("Abort")]
            Abort,
        }

        #endregion

        #region AsyncCommand

        /// <summary>
        /// AsyncCommand Enum for storing command values
        /// </summary>
        public enum AsyncCommand
        {
            /// <summary>Stop</summary>
            [StringValue("Stop")]
            Stop = 1,

            /// <summary>Abort</summary>
            [StringValue("Abort")]
            Abort,
        }

        #endregion
    }

    /// <summary>
    ///  To get the string value
    /// </summary>
    public class StringValueAttribute : System.Attribute
    {
        private string _value;

        /// <summary>StringValueAttribute</summary>
        /// <param name="value">value</param>
        public StringValueAttribute(string value)
        {
            _value = value;
        }

        /// <summary>Value</summary>
        public string Value
        {
            get { return _value; }
        }
    }

    /// <summary>
    ///  Class that holds the Enum values string
    /// </summary>
    public class StringEnum
    {
        /// <summary>
        ///  To get the string value from Enum value
        /// </summary>
        /// <param name="value">Enum value</param>
        /// <returns>String value of Enum</returns>
        public static string GetStringValue(Enum value)
        {
            string output = null;
            Type type = value.GetType();

            // Gets the 'StringValueAttribute'
            FieldInfo fi = type.GetField(value.ToString());
            StringValueAttribute[] attrs =
               fi.GetCustomAttributes(typeof(StringValueAttribute),
                                       false) as StringValueAttribute[];
            if (attrs.Length > 0)
            {
                output = attrs[0].Value;
            }
            return output;
        }
    }
}

File: root/programs/VB/Frameworks/Infrastructure/Business/AsyncProcessingService/AsyncProcessingServiceParameterValue.vb
'**********************************************************************************
'* Copyright (C) 2007,2016 Hitachi Solutions,Ltd.
'**********************************************************************************

#Region "Apache License"
'
' Licensed under the Apache License, Version 2.0 (the "License");
' you may not use this file except in compliance with the License. 
' You may obtain a copy of the License at
'
' http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS,
' WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
' See the License for the specific language governing permissions and
' limitations under the License.
'
#End Region

'**********************************************************************************
'* クラス名        ：AsyncProcessingServiceParameterValue
'* クラス日本語名  ：AsyncProcessingServiceParameterValue
'*
'*  日時        更新者            内容
'*  ----------  ----------------  -------------------------------------------------
'*  11/28/2014  Supragyan         Paramter Value class for Asynchronous Processing Service
'*  04/15/2015  Sandeep           Changed datatype of ProgressRate to decimal.
'**********************************************************************************

Imports System.Reflection

Imports Touryo.Infrastructure.Business.Common
Imports Touryo.Infrastructure.Business.Util

Namespace Touryo.Infrastructure.Business.AsyncProcessingService
	''' <summary>
	''' Paramter Value class for Asynchronous Processing Service
	''' </summary>
	Public Class AsyncProcessingServiceParameterValue
		Inherits MyParameterValue
		''' <summary>汎用エリア</summary>
		Public Obj As Object

		''' <summary>TaskId</summary>
		Public TaskId As Integer

		''' <summary>UserId</summary>
		Public UserId As String

		''' <summary>ProcessName</summary>
		Public ProcessName As String

		''' <summary>Data</summary>
		Public Data As String

		''' <summary>RegistrationDateTime</summary>
		Public RegistrationDateTime As DateTime

		''' <summary>ExecutionStartDateTime</summary>
		Public ExecutionStartDateTime As DateTime

		''' <summary>NumberOfRetries</summary>
		Public NumberOfRetries As Integer

		''' <summary>ProgressRate</summary>
		Public ProgressRate As Decimal

		''' <summary>Status</summary>
		Public StatusId As Integer

		''' <summary>CompletionDateTime</summary>
		Public CompletionDateTime As DateTime

		''' <summary>CommandId</summary>
		Public CommandId As Integer

		''' <summary>ReservedArea</summary>
		Public ReservedArea As String

		''' <summary>ExceptionInfo</summary>
		Public ExceptionInfo As String

		#Region "コンストラクタ"

		''' <summary>コンストラクタ</summary>
		Public Sub New(screenId As String, controlId As String, methodName As String, actionType As String, user As MyUserInfo)
				' Baseのコンストラクタに引数を渡すために必要。
			MyBase.New(screenId, controlId, methodName, actionType, user)
		End Sub

		#End Region

		#Region "AsyncStatus"

		''' <summary>
		''' AsyncStatus Enum for storing all status
		''' </summary>
		Public Enum AsyncStatus
			''' <summary>Register</summary>
			<StringValue("Register")> _
			Register = 1

			''' <summary>Processing</summary>
			<StringValue("Processing")> _
			Processing

			''' <summary>End</summary>
			<StringValue("End")> _
			[End]

			''' <summary>AbnormalEnd</summary>
			<StringValue("AbnormalEnd")> _
			AbnormalEnd

			''' <summary>Abort</summary>
			<StringValue("Abort")> _
			Abort
		End Enum

		#End Region

		#Region "AsyncCommand"

		''' <summary>
		''' AsyncCommand Enum for storing command values
		''' </summary>
		Public Enum AsyncCommand
			''' <summary>Stop</summary>
			<StringValue("Stop")> _
			[Stop] = 1

			''' <summary>Abort</summary>
			<StringValue("Abort")> _
			Abort
		End Enum

		#End Region
	End Class

	''' <summary>
	'''  To get the string value
	''' </summary>
	Public Class StringValueAttribute
		Inherits System.Attribute
		Private _value As String

		''' <summary>StringValueAttribute</summary>
		''' <param name="value">value</param>
		Public Sub New(value As String)
			_value = value
		End Sub

		''' <summary>Value</summary>
		Public ReadOnly Property Value() As String
			Get
				Return _value
			End Get
		End Property
	End Class

	''' <summary>
	'''  Class that holds the Enum values string
	''' </summary>
	Public Class StringEnum
		''' <summary>
		'''  To get the string value from Enum value
		''' </summary>
		''' <param name="value">Enum value</param>
		''' <returns>String value of Enum</returns>
		Public Shared Function GetStringValue(value As [Enum]) As String
			Dim output As String = Nothing
			Dim type As Type = value.[GetType]()

			' Gets the 'StringValueAttribute'
			Dim fi As FieldInfo = type.GetField(value.ToString())
            Dim attrs As StringValueAttribute() = TryCast(fi.GetCustomAttributes(GetType(StringValueAttribute), False), StringValueAttribute())
			If attrs.Length > 0 Then
				output = attrs(0).Value
			End If
			Return output
		End Function
	End Class
End Namespace
